### PR TITLE
Certificate model refactor and tests

### PR DESCRIPTION
**Pull request by d1str0**
Depends on PR #23 

DO NOT MERGE UNTIL REBASE

Test by running 'go test' in model directory. Should pass. 'go test -v' will show all the tests and you should see 3 new ones for certificates.

### CODE REVIEW

File: model/certificate_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+			}
+
+			if c.CertURL != "" {
+				t.Error("cert url should be blank")
+			}
+			if c.CertStableURL != "" {
+				t.Error("cert stable url should be blank")
+			}
+
+			if len(c.PrivateKey) != 0 {
```
Comment (debus) [inline model/certificate_test.go:81]: I assume this passes and I'm missing something, but it looks like `PrivateKey` is filled out by the `NewCertificate` function. So shouldn't this always fail?

Comment (d1str0) [reply model/certificate_test.go:81]: There's a small difference between the ACMEKey (private key) and the actual Certificate's Private Key. This isn't super clear in the actual definition of NewCertificate.

**Comment by d1str0** [comment]
This has been rebased off of #23

File: model/certificate_test.go (hunk 1 above)
Comment (debus) [reply model/certificate_test.go:81]: Oh whoops I see it now. I must have switched the fieldname fieldvalue order around in my head lol

**Review by debus — COMMENTED** [review]
LGTM :+1:

**Pull request merged**